storage/innobase/log/log0recv.cc

```diff
@@ -2847,7 +2847,8 @@ recv_sys_t::parse_mtr_result recv_sys_t::parse(source &l, bool if_exists)
                                    last_offset)
                 : file_name_t::initial_flags;
               if (it == recv_spaces.end())
-                ut_ad(!file_checkpoint || space_id == TRX_SYS_SPACE ||
+                ut_ad(!store ||
+                      !file_checkpoint || space_id == TRX_SYS_SPACE ||
                       srv_is_undo_tablespace(space_id));
```

Comment:
This code fragment is part of the 
`          if (UNIV_UNLIKELY(!page_no) && file_checkpoint)`
So, `!file_checkpoint ` can never be true here and is redundant.

Reply:
Good point. Fixed.